test/src/main/java/org/openjdk/skara/test/TestHostedRepository.java

```diff
@@ -55,6 +55,15 @@ public TestHostedRepository(TestHost host, String projectName, Repository localR
         commitComments = new HashMap<>();
     }
 
+    public TestHostedRepository(TestHost host, String projectName) {
```

Comment:
Could you add a comment explaining the limitations of creating an instance with this constructor? Something like:
"Creates an instance without a backing local repository that will not support any actual repository interaction."

Also since we are always calling this with host=null, we might as well remove that parameter too.